src/v/storage/log_manager.cc

```diff
@@ -401,6 +402,69 @@ ss::future<> log_manager::remove(model::ntp ntp) {
     });
 }
 
+/// Parse partition directory name
+static std::optional<std::pair<model::partition_id, model::revision_id>>
+parse_partition_directory(const ss::sstring& name) {
+    const std::regex re(R"(^(\d+)_(\d+)$)");
+    std::cmatch match;
+    if (!std::regex_match(name.c_str(), match, re)) {
+        return std::nullopt;
+    }
+    return std::make_pair(
+      model::partition_id(boost::lexical_cast<uint64_t>(match[1].str())),
+      model::revision_id(boost::lexical_cast<uint64_t>(match[2].str())));
+}
+
+ss::future<> log_manager::remove_orphan(
+  ss::sstring topic_directory_path, model::ntp ntp, model::revision_id rev) {
```

Comment:
i think we do not need to provide topic directory here as we have an `ntp` already, it should be enough to provide base data directory